website/server/libs/stringUtils.js

```diff
@@ -4,12 +4,20 @@ export function removePunctuationFromString (str) {
 
 // NOTE: the wordsToMatch aren't escaped in order to support regular expressions,
 // so this method should not be used if wordsToMatch contains unsanitized user input
+
 export function getMatchesByWordArray (str, wordsToMatch) {
+  // remove accented characters from the string, which would trip up the regEx
```

Comment:
@Alys what do you think about this fix where accented characters are simply ignored?

Reply:
I llke this approach. I think it will work well for our needs. 

@bigsee Thanks for including those two URLS - they were useful.

Reply:
@paglias just FYI, I don't think they are technically ignored. they are converted (for validation only) into their non-accented form. If you're happy I'll get on those extra tests.

Reply:
I said ignored because `.replace(/[\u0300-\u036f]/g, '');` doesn't replace the chars from u0300 to u036f with an empty char?

Although in this test it didn't seem to change anything:

<img width="545" alt="Schermata 2020-08-17 alle 12 15 38" src="https://user-images.githubusercontent.com/589911/90385407-63ec7580-e083-11ea-9fe1-bcdede22c3b5.png">

Reply:
I am only learning this stuff myself so please excuse me if I have something wrong or explain something with inaccurate terms. :) 

I think you are missing the 'NFD' argument in the normalize() function:

![Screen Shot 2020-08-17 at 12 45 04](https://user-images.githubusercontent.com/11768967/90388120-9e581180-e087-11ea-9820-d20599380427.png)

You are correct that the code replaces the chars from u0300 to u036f with an empty char. However, this Unicode range is a range of _diacritical marks_ rather than accented characters. So I would not expect any accented characters to get replaced. I think the accented character are somehow the _combination_ of the non-accented character with a diacritical mark...

...and I think that without this "normalization"/conversion first, `SOMESWEARWORDé` is evaluated as the split version => `SOMESWEARWORD [diacritical mark] e` somehow. So `str.normalize('NFD').replace(/[\u0300-\u036f]/g, '')` aims to remove those diacritical marks and effectively treat the input as one complete string again.

Not sure fully what is happening under the hood but that's my layperson's guess from what I read.

Reply:
Oh you're right sorry! I thought NFD was the default parameter when using `normalize` so I skipped it. And unicode is something I definitely need to read more about, your explanation makes total sense :)